hoprd/rest-api/src/types.rs

```diff
@@ -0,0 +1,101 @@
+// Unified type for PeerId and Address
+//
+// This module provides a unified type for PeerId and Address. This is useful for APIs that accept both PeerId and Address.
+
+use crate::ApiErrorStatus;
+use axum::http::StatusCode;
+use core::result::Result;
+use hopr_crypto_types::types::OffchainPublicKey;
+use hopr_db_sql::prelude::HoprDbResolverOperations;
+use hopr_lib::{Address, GeneralError};
+use libp2p_identity::PeerId;
+use serde::{Deserialize, Serialize};
+use std::fmt::{Debug, Display, Formatter};
+use std::str::FromStr;
+
+#[derive(Clone, Copy, Default, Eq, Hash, Ord, Serialize, Deserialize, PartialEq, PartialOrd)]
+pub struct PeerOrAddress {
+    pub peer_id: Option<PeerId>,
+    pub address: Option<Address>,
+}
+
+impl PeerOrAddress {
+    pub fn new(identifier: String) -> Result<Self, GeneralError> {
+        if let Ok(peer_id) = PeerId::from_str(&identifier) {
+            Ok(Self::from(peer_id))
+        } else if let Ok(address) = Address::from_str(&identifier) {
+            Ok(Self::from(address))
+        } else {
+            Err(GeneralError::InvalidInput)
+        }
+    }
+
+    pub async fn fulfill<T: HoprDbResolverOperations>(
+        &mut self,
+        resolver: &T,
+    ) -> Result<Self, (StatusCode, ApiErrorStatus)> {
```

Comment:
This method should return a some custom error, which is later mapped to `(StatusCode, ApiErrorStatus)` by the caller.

As it is now, it makes the `PeerOrAddress` is less universal.